src/client/Client.cc

```diff
@@ -1539,6 +1539,24 @@ void Client::connect_mds_targets(mds_rank_t mds)
   }
 }
 
+void Client::connect_mds_targets(mds_rank_t mds, std::unique_ptr<MDSMap>& newmap)
+{
+  ldout(cct, 10) << __func__ << " for mds." << mds << dendl;
+  ceph_assert(mds_sessions.count(mds));
+
+  const MDSMap::mds_info_t& info = newmap->get_mds_info(mds);
+  for (set<mds_rank_t>::const_iterator q = info.export_targets.begin();
+       q != info.export_targets.end();
+       ++q) {
```

Comment:
In new code, I discourage the use of iterators when possible.
```suggestion
  for (const auto& rank : info.export_targets) {
```